src/core/video/video.service.ts

```diff
@@ -0,0 +1,58 @@
+import type { ResizeOptions } from 'sharp';
+import type { Readable, Stream } from 'stream';
+import ffmpegBinary from '@ffmpeg-installer/ffmpeg';
+import type { FfprobeData } from 'fluent-ffmpeg';
+import ffmpeg from 'fluent-ffmpeg';
+import fs from 'fs';
+import tmp from 'tmp';
+import util from 'util';
+
+const createTempFilename = util.promisify(tmp.tmpName);
+
+type VideoResizeOptions = {
+  codecName: string;
+} & ResizeOptions
+
+export class VideoService {
+  constructor() {
+    ffmpeg.setFfmpegPath(ffmpegBinary.path);
+  }
+
+  public resize(file: Readable, options: VideoResizeOptions): Stream {
+    return ffmpeg()
+      .input(file)
+      .size(`${options.height}x${options.width}`)
+      .format(options.codecName)
+      .pipe();
+  }
+
+  public metadata(file: Readable): Promise<FfprobeData> {
+    return new Promise((resolve, reject) => {
+      ffmpeg().input(file).ffprobe((err, data) => {
+        if (err) {
+          reject(err);
+        }
+
+        resolve(data);
+      });
+    });
+  }
+
+  public async thumbnail(file: Readable, options: { second: number }): Promise<Stream> {
+    const tempFileName = `${await createTempFilename()}.png`;
```

Comment:
According to docs: https://github.com/fluent-ffmpeg/node-fluent-ffmpeg#screenshotsoptions-dirname-generate-thumbnails
it doesn't work on input streams. That's why I used temporary file - it should be small enough to work pretty fast.

Reply:
Or maybe you have some examples on how did you resolve that in other project?

Reply:
That's ok imho